main.tf

```diff
@@ -15,14 +32,29 @@ resource "aws_security_group" "this" {
   }
 }
 
+##################################
+# Get ID of created Security Group
+##################################
+locals {
+  this_sg_id = "${
```

Comment:
This is correct when SG creation is disabled:
`${element(concat(coalescelist(aws_security_group.this.*.id, aws_security_group.this_name_prefix.*.id), list("")), 0)}`

Also, write this in one line.

Reply:
Also, move `locals` block at the top on the file.